s3/pkg/service/object.go

```diff
@@ -52,6 +52,58 @@ func (s *s3Service) UpdateObject(ctx context.Context, in *pb.Object, out *pb.Bas
 	return nil
 }
 
+func (s *s3Service) RestoreObject(ctx context.Context, req *pb.RestoreObjectRequest, out *pb.BaseResponse) error {
+	log.Infoln("RestoreObject is called in s3 service.")
+	bucketName := req.Restore.BucketName
+	objectName := req.Restore.ObjectKey
+
+	var err error
+
+	object, err := s.MetaStorage.GetObject(ctx, bucketName, objectName, "", true)
+	if err != nil {
+		log.Errorln("failed to get object info from meta storage. err:", err)
+		return err
+	}
+
+	bucket, err := s.MetaStorage.GetBucket(ctx, bucketName, true)
+	if err != nil {
+		log.Errorln("failed to get bucket from meta storage. err:", err)
+		return err
+	}
+
+	isAdmin, tenantId, _, err := util.GetCredentialFromCtx(ctx)
+	if err != nil && isAdmin == false {
+		log.Error("get tenant id failed")
+		err = ErrInternalError
```

Comment:
IMO error code an be "unauthorized" instead of internal error

Reply:
This is not unauthorized. Actually its an issue getting tenant id at first place